ompi/runtime/ompi_spc.c

```diff
@@ -247,26 +394,339 @@ static int ompi_spc_get_count(const struct mca_base_pvar_t *pvar, void *value, v
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_REDUCE_LINEAR);
+
+    /* Reduce Scatter Algorithms */
+    /* Collective bin counter for the Nonoverlapping Reduce Scatter algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_REDUCE_SCATTER_NONOVERLAPPING);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_REDUCE_SCATTER_NONOVERLAPPING);
+    /* Collective bin counter for the Recursive Halving Reduce Scatter algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_REDUCE_SCATTER_RECURSIVE_HALVING);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_REDUCE_SCATTER_RECURSIVE_HALVING);
+    /* Collective bin counter for the Ring Reduce Scatter algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_REDUCE_SCATTER_RING);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_REDUCE_SCATTER_RING);
+
+    /* Scatter Algorithms */
+    /* Collective bin counter for the Binomial Scatter algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_SCATTER_BINOMIAL);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_SCATTER_BINOMIAL);
+    /* Collective bin counter for the Linear Scatter algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_SCATTER_LINEAR);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_SCATTER_LINEAR);
+
+#if 0
+    /* X Algorithms */
+    /* Collective bin counter for the X algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_);
+    /* Collective bin counter for the X algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_);
+    /* Collective bin counter for the X algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_);
+    /* Collective bin counter for the X algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_);
+    /* Collective bin counter for the X algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_);
+    /* Collective bin counter for the X algorithm */
+    SET_SPC_BIT(ompi_spc_collective_bin_event, OMPI_SPC_BASE_);
+    SET_SPC_BIT(ompi_spc_bin_event, OMPI_SPC_BASE_);
+#endif
+
+    for(i = 0; i < OMPI_SPC_NUM_COUNTERS; i++) {
+        if(IS_SPC_BIT_SET(ompi_spc_collective_bin_event,i)) {
+            data_size += 4 * (sizeof(int) + sizeof(ompi_spc_value_t));
+            ompi_spc_offsets[i].num_bins = 4;
+        }
+    }
+
+    /* ###############################################################################
+     * ###############################################################################
+     * ###############################################################################
+     */
+
+    int bytes_needed = PAGE_SIZE;
```

Comment:
Really ? You need a while loop to compute the number of pages needed for certain amount of data ?

How about ```bytes_needed = PAGE_SIZE * ((data_size  + PAGE_SIZE - 1) % PAGE_SIZE);``` ?

Reply:
Fixed